crates/sui-core/src/authority/consensus_tx_status_cache.rs

```diff
@@ -160,6 +191,10 @@ impl ConsensusTxStatusCache {
 
     pub async fn update_last_committed_leader_round(&self, leader_round: u32) {
```

Comment:
I am thinking that ideally we would like the  `gc_round` to be passed as parameter to this method and basically be provided by consensus it self (ex maybe as part of the committed sub dag, or a method there etc). Inside consensus we deliberately tried to be consistent on having one source of truth for this (DagState) rather than computing in different places (like we did in the past with Narwhal, which was a pain at some point to maintain). I know that this could trigger more refactoring, but leaving it as comment in case you also share similar thoughts/concerns.